lucille-core/src/test/java/com/kmwllc/lucille/stage/ParseFloatsTest.java

```diff
@@ -0,0 +1,52 @@
+package com.kmwllc.lucille.stage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import java.util.List;
+import org.junit.Test;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+
+public class ParseFloatsTest {
+
+  private StageFactory factory = StageFactory.of(ParseFloats.class);
+
+  @Test
+  public void testMalformedConfig() {
+    assertThrows(StageException.class, () -> factory.get("ParseFloatsTest/bad.conf"));
+  }
+
+  @Test
+  public void testProcessDocument() throws StageException {
+    Stage stage = factory.get("ParseFloatsTest/basic.conf");
+
+    Document bad = Document.create("doc1");
+    Document empty = Document.create("doc2");
+    Document simple = Document.create("doc3");
+    Document noField = Document.create("doc4");
+
+    bad.setOrAdd("source", "not a float");
+    empty.setOrAdd("source", "[]");
+    simple.setOrAdd("source", "[1, 2, 3]");
```

Comment:
for completeness, I'd suggest giving one of the floats a decimal point... like 3.5 instead of 3
and adding a separate doc where the array has mostly floats but one bad value, i.e. [1, 2.3, "a", 4]